realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,729 @@
+     * @return the zero-based column index for the given column name, or -1 if
+     * the column name does not exist.
+     * @see #getColumnIndexOrThrow(String)
+     */
+    @Override
+    public int getColumnIndex(String columnName) {
+        checkClosed();
+        if (DEFAULT_ID_COLUMN.equals(columnName)) {
+            return (int) idColumnIndex;
+        }
+        return (int) table.getColumnIndex(columnName);
+    }
+
+    /**
+     * Returns the zero-based index for the given column name, or throws
+     * {@link IllegalArgumentException} if the column doesn't exist. If you're not sure if
+     * a column will exist or not use {@link #getColumnIndex(String)} and check for -1, which
+     * is more efficient than catching the exceptions.
+     *
+     * @param columnName the name of the target column.
+     * @return the zero-based column index for the given column name
+     * @see #getColumnIndex(String)
+     * @throws IllegalArgumentException if the column does not exist
+     */
+    @Override
+    public int getColumnIndexOrThrow(String columnName) throws IllegalArgumentException {
+        checkClosed();
+        int index = (int) table.getColumnIndex(columnName);
+        if (index == TableOrView.NO_MATCH) {
+            throw new IllegalArgumentException(columnName + " not found in this cursor.");
+        }
+        return index;
+    }
+
+    /**
+     * Returns the column name at the given zero-based column index.
+     *
+     * @param columnIndex the zero-based index of the target column.
+     * @return the column name for the given column index.
+     */
+    @Override
+    public String getColumnName(int columnIndex) {
+        checkClosed();
+        return table.getColumnName(columnIndex);
+    }
+
+    /**
+     * Returns a string array holding the names of all of the columns in the
+     * result set in the order in which they were listed in the result.
+     *
+     * @return the names of the columns returned in this query.
+     */
+    @Override
+    public String[] getColumnNames() {
+        checkClosed();
+        int columns = (int) table.getColumnCount();
+        String[] columnNames = new String[columns];
+        for (int i = 0; i < columns; i++) {
+            columnNames[i] = table.getColumnName(i);
+        }
+        return columnNames;
+    }
+
+    /**
+     * Return total number of columns
+     * @return number of columns
+     */
+    @Override
+    public int getColumnCount() {
+        checkClosed();
+        return (int) table.getColumnCount();
+    }
+
+    /**
+     * Returns the value of the requested column as a byte array.
+     *
+     * @param columnIndex the zero-based index of the target column.
+     * @return the value of that column as a byte array.
+     * @throws IllegalArgumentException if the requested column is not a byte array.
+     */
+    @Override
+    public byte[] getBlob(int columnIndex) {
+        checkClosed();
+        checkPosition();
```

Comment:
Do you ever call checkPosition() without also checkClosed()?
Consider making a checkClosedAndPosition()